source/code/source/playfab/PlayFabCurlHttpPlugin.cpp

```diff
@@ -110,6 +112,21 @@ namespace PlayFab
         } // UNLOCK httpRequestMutex
     }
 
+    constexpr char requestIdHeaderKey[] = "X-RequestId: ";
+    size_t HeaderCallback(char* buffer, size_t size, size_t nitems, void* userdata)
+    {
+        size_t headerLen = _countof(requestIdHeaderKey) - 1;
+        CallRequestContainer& reqContainer = *static_cast<CallRequestContainer*>(userdata);
+        // If this header starts with the key we expect
+        if (strncasecmp(buffer, requestIdHeaderKey, headerLen) == 0)
```

Comment:
>strncasecmp(buffer, requestIdHeaderKey, headerLen) [](start = 12, length = 50)

Can't assume that the string passed in the buffer is always at least headerLen long (it can be shorter! and not necessarily end with 0!). I think you may also need to use size and nitems arguments here somehow to make it safe.

Reply:
Added a length test, which should prevent any potential buffer-overrun.